exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/partitionsender/PartitionerTemplate.java

```diff
@@ -376,9 +376,8 @@ public void updateStats(FragmentWritableBatch writableBatch) {
      */
     public void initializeBatch() {
       for (VectorWrapper<?> v : incoming) {
-        // create new vector
-        @SuppressWarnings("resource")
-        ValueVector outgoingVector = TypeHelper.getNewVector(v.getField(), allocator);
+        // create new vector by cloning the incoming vector's type
+        ValueVector outgoingVector = TypeHelper.getClonedVector(v.getField(), allocator);
```

Comment:
This is the heart of the change: this is the place where we were reusing previously-owned materialized fields.